modules/dns_response_policy/main.tf

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+Cloud DNS private zones let you create a single response policy per network that
+modifies resolver behavior according to the policy created.
+*/
+resource "google_dns_response_policy" "this" {
+  provider             = google-beta
+  project              = var.project_id
+  response_policy_name = var.policy_name
+  description          = var.description
+  dynamic "networks" {
+    for_each = var.network_self_links
+    content {
+      network_url = networks.value
+    }
+  }
+}
+
+/**
+Response policy rules can be created under a response policy to alter results for
+selected query names or trigger passthru behavior that bypasses the response policy.
+*/
+resource "google_dns_response_policy_rule" "this" {
+  for_each        = toset(keys(var.rules))
+  provider        = google-beta
+  project         = var.project_id
+  rule_name       = each.key
+  dns_name        = lookup(var.rules[each.key], "dns_name")
+  response_policy = google_dns_response_policy.this.response_policy_name
+  behavior        = lookup(var.rules[each.key], "rule_behavior", "") == "BYPASS" ? "bypassResponsePolicy" : null
```

Comment:
In order to keep it simple can we use to values allowed by the [provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dns_response_policy_rule#behavior) `bypassResponsePolicy`? It also allows `behaviorUnspecified` and may be more values in future.

Reply:
Yes lets create an issue for this error on provider site. I am not sure if it is a provider issue or API.

Reply:
ack, let me do that with an error message. I'll reproduce this issue once.